src/utils/messaging-outbox/src/entities/outbox_message.rs

```diff
@@ -13,12 +13,36 @@ use crate::OutboxMessageID;
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
+// Version of outbox message which will be bumped only if the structure contains
+// breaking changes in the message structure
+pub const OUTBOX_MESSAGE_VERSION: i32 = 1;
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
 #[derive(Debug, Clone)]
 pub struct OutboxMessage {
     pub message_id: OutboxMessageID,
     pub producer_name: String,
     pub content_json: serde_json::Value,
     pub occurred_on: DateTime<Utc>,
+    pub version: i32,
```

Comment:
u32? I guess we don't need negative versions

Reply:
It's kind tricky, because we need to perform additional cast to support postgres insert. They removed support of u32. I would keep as it is. WDYT?

Reply:
Got it, thanks a lot for providing the context. Yes, you are right, I recall problems with Postgres :( It's a shame that one of the implementations imposes a type for us within the model.

Still, it doesn't look like (to me) a worthwhile reason to complicate the decision. For this reason, please add a short comment to the field why `i32` is chosen, otherwise the readers eye will really cling to it

Reply:
Personally I would any day prefer a bunch ugly type casts on repository implementation level to imprecise types at domain layer. Serialization should not dictate us the types. But I think we already made this trade-off in `OutboxMessageID` and perhaps other ID types, so won't insist in this simple case, but in general good domain types should make invalid state unrepresentable.

Reply:
Totally true. Besides. we have a lot of ugly casts in repository already, so a few more won't change the picture there, but the model would better  be clean.